### PR TITLE
[v6r15] Fixed Jenkins alerts on ConsistencyInspector.py

### PR DESCRIPTION
**Pull request by marcelovilaca**
Fixed Jenkins alerts on ConsistencyInspector.py and added method __getDirectories(self)

### CODE REVIEW

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 1]
```diff
@@ -561,7 +582,7 @@ def __checkPhysicalFiles( self, replicas, catalogMetadata, ses = None ):
           if ( metadata['Size'] != catalogMetadata[lfn]['Size'] ):                # and ( metadata['Size'] != 0 ):
             sizeMismatch.append( ( lfn, 'deprecatedUrl', se, 'CatalogPFNSizeMismatch' ) )
       if sizeMismatch:
-        self.__reportProblematicReplicas( sizeMismatch, se, 'CatalogPFNSizeMismatch' )
+        self.dic.__reportProblematicReplicas( sizeMismatch, se, 'CatalogPFNSizeMismatch' )
```
Comment (fstagni) [inline DataManagementSystem/Client/ConsistencyInspector.py]: You can't access private methods. And this is true also for the other changes.

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 2]
```diff
@@ -235,6 +235,27 @@ def _getTSFiles( self ):
...
+    if self.directories:
+      directories = []
+      printout = False
+      for directory in self.directories:
+        if not directory.endswith( '...' ):
+          directories.append( directory )
+        else:
+          printout = True
+          topDir = os.path.dirname( directory )
+          res = self.dm.getCatalogListDirectory( topDir )
```
Comment (atsareg) [inline DataManagementSystem/Client/ConsistencyInspector.py]: DataManager does not have this method any more. Use FileCatalog() object to get the directory listing

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 3]
```diff
@@ -80,9 +81,10 @@ def __init__( self, interactive = True, transClient = None, dm = None, fc = None
   ################################################################################
 
   def getReplicasPresence( self, lfns ):
-    """ get the replicas using the standard DataManager.getReplicas()
+    """ get the replicas using the standard FileCatalog.getReplicas()
     """
     lfns = checkArgumentFormat(lfns)
+    print lfns
```
Comment (atsareg) [inline DataManagementSystem/Client/ConsistencyInspector.py]: ??? Some debugging printout ?

Comment (marcelovilaca) [inline DataManagementSystem/Client/ConsistencyInspector.py]: Oops, sorry....

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 4]
```diff
@@ -235,6 +236,28 @@ def _getTSFiles( self ):
...
+          res = self.fc.listDirectory( topDir )
+          #res = self.dm.getCatalogListDirectory( topDir )
+          if not res['OK']:
+            raise RuntimeError( res['Message'] )
+          else:
+            matchDir = directory.split( '...' )[0]
+            directories += [d for d in res['Value']['Successful'].get( topDir, {} ).get( 'SubDirs', [] ) if d.startswith( matchDir )]
+      if printout:
+        gLogger.always( 'Expanded list of %d directories:\n%s' % ( len( directories ), '\n'.join( directories ) ) )
+      return directories
```
Comment (atsareg) [inline DataManagementSystem/Client/ConsistencyInspector.py]: if self.directories is empty then this function returns None, in this case the caller of this function will fail

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 5]
```diff
@@ -235,6 +236,29 @@ def _getTSFiles( self ):
...
+    else:                                                     
+      raise RuntimeError( "Need to specify the directories" )
```
Comment (atsareg) [inline DataManagementSystem/Client/ConsistencyInspector.py]: In DIRAC we do not use Exceptions except some special cases. Either return S_OK/S_ERROR(DError) or just return an empty list.

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 6]
```diff
@@ -467,7 +490,7 @@ def get_fileTypesExcluded( self ):
 
   def set_lfns( self, value ):
     """ Setter """
-    if type( value ) == type( "" ):
+    if isinstance( value , "" ):
```
Comment (fstagni) [inline DataManagementSystem/Client/ConsistencyInspector.py]: if isinstance( value , basestr )

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 7]
```diff
@@ -490,7 +513,7 @@ def catalogDirectoryToSE( self, lfnDir ):
     gLogger.info( "-" * 40 )
     gLogger.info( "Performing the FC->SE check" )
     gLogger.info( "-" * 40 )
-    if type( lfnDir ) in types.StringTypes:
+    if isinstance( lfnDir, types.StringTypes):
```
Comment (fstagni) [inline DataManagementSystem/Client/ConsistencyInspector.py]: if isinstance( lfnDir, basestr):

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 8]
```diff
@@ -509,7 +532,7 @@ def catalogFileToSE( self, lfns ):
     gLogger.info( "-" * 40 )
     gLogger.info( "Performing the FC->SE check" )
     gLogger.info( "-" * 40 )
-    if type( lfns ) in types.StringTypes:
+    if isinstance( lfns, types.StringTypes):
```
Comment (fstagni) [inline DataManagementSystem/Client/ConsistencyInspector.py]: if isinstance( lfns, basestr):

**Comment by fstagni** [comment]
Please replace .items() with .iteritems()

**Comment by chaen** [comment]
Just to give a +1 to that comment, but be careful doing it :-)

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 9]
```diff
@@ -319,10 +343,10 @@ def compareChecksum( self, lfns ):
       self.__write( "Get replicas for %d files (chunks of %d): " % ( len( lfnsLeft ), chunkSize ) )
       for lfnChunk in breakListIntoChunks( lfnsLeft, chunkSize ):
         self.__write( '.' )
-        replicasRes = self.dm.getReplicas( lfnChunk )
+        replicasRes = self.fc.getReplicas( lfnChunk )
         if not replicasRes['OK']:
           gLogger.error( "error:  %s" % replicasRes['Message'] )
-          S_ERROR("error:  %s" % replicasRes['Message'])          # raise RuntimeError( "error:  %s" % replicasRes['Message'] )
+          S_ERROR("error:  %s" % replicasRes['Message'])
```
Comment (fstagni) [inline DataManagementSystem/Client/ConsistencyInspector.py]: aren't you missing a return statement here?

File: DataManagementSystem/Client/ConsistencyInspector.py [hunk 10]
```diff
@@ -334,9 +358,8 @@ def compareChecksum( self, lfns ):
     for lfnChunk in breakListIntoChunks( replicas.keys(), chunkSize ):
       self.__write( '.' )
       res = self.fc.getFileMetadata( lfnChunk )
-      print "res", res
       if not res['OK']:
-        S_ERROR("error %s" % res['Message'])        # raise RuntimeError( "error %s" % res['Message'] )
+        S_ERROR("error %s" % res['Message'])
```
Comment (fstagni) [inline DataManagementSystem/Client/ConsistencyInspector.py]: Also here

**Comment by fstagni** [comment]
If methods now return S_ERROR instead of raising an exception, then they should also return S_OK(result) instead of just result. And callers of these methods should be aware of the changes.

**Comment by fstagni** [comment]
Code seems ok now, but this clearly would need extensive tests. Also, please run pylint on it.

Is it ready to be merged?

**Comment by atsareg** [comment]
Review OK

**Pull request merged**